<a href="https://colab.research.google.com/github/mskinfo/financial-advisor/blob/main/vix_trading.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [ ]:
vixdata = pd.read_excel(r"/content/vix_data.xlsx")
vixdata

DATE   OPEN   HIGH    LOW  CLOSE  T1OPEN  T1HIGH  T1LOW  T1CLOSE  \
0    1990-01-08  20.26  20.26  20.26  20.26   20.11   20.11  20.11    20.11   
1    1990-01-09  22.20  22.20  22.20  22.20   20.26   20.26  20.26    20.26   
2    1990-01-10  22.44  22.44  22.44  22.44   22.20   22.20  22.20    22.20   
3    1990-01-11  20.05  20.05  20.05  20.05   22.44   22.44  22.44    22.44   
4    1990-01-12  24.64  24.64  24.64  24.64   20.05   20.05  20.05    20.05   
...         ...    ...    ...    ...    ...     ...     ...    ...      ...   
8283 2022-11-21  24.05  24.12  22.30  22.36   24.03   24.12  22.98    23.12   
8284 2022-11-22  22.59  22.71  21.28  21.29   24.05   24.12  22.30    22.36   
8285 2022-11-23  21.49  21.78  20.32  20.35   22.59   22.71  21.28    21.29   
8286 2022-11-24  20.50  20.54  20.31  20.42   21.49   21.78  20.32    20.35   
8287 2022-11-25  20.61  21.11  20.46  20.50   20.50   20.54  20.31    20.42   

      T2OPEN  ...  T2LOW  T2CLOSE  T3OPEN  T3HIGH  T3LOW  T3CLOSE  T4OPEN  \
0      19.22  ...  19.22    19.22   18.19   18.19  18.19    18.19   17.24   
1      20.11  ...  20.11    20.11   19.22   19.22  19.22    19.22   18.19   
2      20.26  ...  20.26    20.26   20.11   20.11  20.11    20.11   19.22   
3      22.20  ...  22.20    22.20   20.26   20.26  20.26    20.26   20.11   
4      22.44  ...  22.44    22.44   22.20   22.20  22.20    22.20   20.26   
...      ...  ...    ...      ...     ...     ...    ...      ...     ...   
8283   24.05  ...  23.81    23.93   24.29   24.72  23.99    24.11   23.77   
8284   24.03  ...  22.98    23.12   24.05   25.18  23.81    23.93   24.29   
8285   24.05  ...  22.30    22.36   24.03   24.12  22.98    23.12   24.05   
8286   22.59  ...  21.28    21.29   24.05   24.12  22.30    22.36   24.03   
8287   21.49  ...  20.32    20.35   22.59   22.71  21.28    21.29   24.05   

      T4HIGH  T4LOW  T4CLOSE  
0      17.24  17.24    17.24  
1      18.19  18.19    18.19  
2      19.22  19.22    19.22  
3      20.11  20.11    20.11  
4      20.26  20.26    20.26  
...      ...    ...      ...  
8283   26.22  23.18    24.54  
8284   24.72  23.99    24.11  
8285   25.18  23.81    23.93  
8286   24.12  22.98    23.12  
8287   24.12  22.30    22.36  

[8288 rows x 21 columns]

In [ ]:
from sklearn import linear_model
X = vixdata[['T1OPEN', 'T1HIGH', 'T1LOW', 'T1CLOSE', 'T2OPEN', 'T2HIGH', 'T2LOW', 'T2CLOSE', 'T3OPEN', 'T3HIGH', 'T3LOW', 'T3CLOSE', 'T4OPEN', 'T4HIGH', 'T4LOW', 'T4CLOSE']]
y = vixdata['CLOSE']

In [22]:
train_x = X[7000:8000]
train_x

T1OPEN  T1HIGH  T1LOW  T1CLOSE  T2OPEN  T2HIGH  T2LOW  T2CLOSE  T3OPEN  \
7000    9.92   10.04   9.29     9.97   10.22   11.77   9.99    10.05   10.34   
7001   10.25   11.08   9.94    11.07    9.92   10.04   9.29     9.97   10.22   
7002   10.89   11.16  10.39    11.16   10.25   11.08   9.94    11.07    9.92   
7003   11.34   13.20  10.99    11.23   10.89   11.16  10.39    11.16   10.25   
7004   11.17   11.81  10.60    11.30   11.34   13.20  10.99    11.23   10.89   
...      ...     ...    ...      ...     ...     ...    ...      ...     ...   
7995   22.92   23.17  20.62    21.30   22.90   24.58  21.88    22.96   24.78   
7996   22.95   24.40  20.99    21.00   22.92   23.17  20.62    21.30   22.90   
7997   20.54   20.76  19.07    19.54   22.95   24.40  20.99    21.00   22.92   
7998   19.46   19.94  18.20    18.77   20.54   20.76  19.07    19.54   22.95   
7999   19.93   20.45  18.11    20.00   19.46   19.94  18.20    18.77   20.54   

      T3HIGH  T3LOW  T3CLOSE  T4OPEN  T4HIGH  T4LOW  T4CLOSE  
7000   10.41   9.87    10.07    9.85   10.46   9.78    10.31  
7001   11.77   9.99    10.05   10.34   10.41   9.87    10.07  
7002   10.04   9.29     9.97   10.22   11.77   9.99    10.05  
7003   11.08   9.94    11.07    9.92   10.04   9.29     9.97  
7004   11.16  10.39    11.16   10.25   11.08   9.94    11.07  
...      ...    ...      ...     ...     ...    ...      ...  
7995   24.89  20.64    21.15   21.48   24.71  20.60    23.14  
7996   24.58  21.88    22.96   24.78   24.89  20.64    21.15  
7997   23.17  20.62    21.30   22.90   24.58  21.88    22.96  
7998   24.40  20.99    21.00   22.92   23.17  20.62    21.30  
7999   20.76  19.07    19.54   22.95   24.40  20.99    21.00  

[1000 rows x 16 columns]

In [23]:
train_y = y[7000:8000]
train_y

7000    11.07
7001    11.16
7002    11.23
7003    11.30
7004     9.80
        ...  
7995    21.00
7996    19.54
7997    18.77
7998    20.00
7999    19.85
Name: CLOSE, Length: 1000, dtype: float64

In [24]:
test_x = X[8000:]
test_x

T1OPEN  T1HIGH  T1LOW  T1CLOSE  T2OPEN  T2HIGH  T2LOW  T2CLOSE  T3OPEN  \
8000   20.62   20.81  18.97    19.85   19.93   20.45  18.11    20.00   19.46   
8001   20.14   20.23  18.44    18.64   20.62   20.81  18.97    19.85   19.93   
8002   18.01   18.08  16.80    16.86   20.14   20.23  18.44    18.64   20.62   
8003   16.64   16.85  15.72    16.30   18.01   18.08  16.80    16.86   20.14   
8004   17.29   17.93  16.27    16.31   16.64   16.85  15.72    16.30   18.01   
...      ...     ...    ...      ...     ...     ...    ...      ...     ...   
8283   24.03   24.12  22.98    23.12   24.05   25.18  23.81    23.93   24.29   
8284   24.05   24.12  22.30    22.36   24.03   24.12  22.98    23.12   24.05   
8285   22.59   22.71  21.28    21.29   24.05   24.12  22.30    22.36   24.03   
8286   21.49   21.78  20.32    20.35   22.59   22.71  21.28    21.29   24.05   
8287   20.50   20.54  20.31    20.42   21.49   21.78  20.32    20.35   22.59   

      T3HIGH  T3LOW  T3CLOSE  T4OPEN  T4HIGH  T4LOW  T4CLOSE  
8000   19.94  18.20    18.77   20.54   20.76  19.07    19.54  
8001   20.45  18.11    20.00   19.46   19.94  18.20    18.77  
8002   20.81  18.97    19.85   19.93   20.45  18.11    20.00  
8003   20.23  18.44    18.64   20.62   20.81  18.97    19.85  
8004   18.08  16.80    16.86   20.14   20.23  18.44    18.64  
...      ...    ...      ...     ...     ...    ...      ...  
8283   24.72  23.99    24.11   23.77   26.22  23.18    24.54  
8284   25.18  23.81    23.93   24.29   24.72  23.99    24.11  
8285   24.12  22.98    23.12   24.05   25.18  23.81    23.93  
8286   24.12  22.30    22.36   24.03   24.12  22.98    23.12  
8287   22.71  21.28    21.29   24.05   24.12  22.30    22.36  

[288 rows x 16 columns]

In [25]:
test_y = y[8000:]
test_y

8000    18.64
8001    16.86
8002    16.30
8003    16.31
8004    15.70
        ...  
8283    22.36
8284    21.29
8285    20.35
8286    20.42
8287    20.50
Name: CLOSE, Length: 288, dtype: float64

In [26]:
regr = linear_model.LinearRegression()
regr.fit(train_x, train_y)

predictions = regr.predict(test_x)
preds = pd.DataFrame(predictions)
test_y = pd.DataFrame(test_y)

test_y.to_excel("testout.xlsx")
preds.to_excel("predsout.xlsx")

# **Theory**
This round of analysis used 1000 days of historical cboe vix data to create a model, which, using historical inputs, attempts to predict the change between the previous close and the next day's close for 288 days.

The model predicts the price direction with an accuracy of 52.621% in terms of the binary direction.

If an agent purchased or shorted the vix based on the predicted binary direction, the average return per day is 0.21%.  Annually, this is 68.95%.

Due to the wash-sale rule, implementing this strategy would be a flawed decision.  The wash-sale rule states that if an agent closes a trade at a loss and opens the same or a substantially identical trade within 30 calendar days before or after the first trade, they will not be able to take a loss for that security on your their tax return.

To avoid this rule, the agent could use futures, or options on futures for the underlying security, as the wash-sale rule does not apply to these instruments.

# **Implementation**
Commissions on options are the only variable cost, at USD 0.65 per trade.  There will be two trades per day, making the monthly trading costs equivalent to $26.  From here, options will magnify returns by a factor of 10.  So, each day, the strategy will yield 0.21% * 10 = 2.1%.

Within the first year, starting with USD 500 risked, this will yield USD 90,241.

Automating this stratey will involving creating a continuous dataframe and using the etrade API to execute trades.